system/launchd.py

```diff
@@ -0,0 +1,405 @@
+
+    return (rc, out, err)
+
+def handle_unloaded(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if launchctl.is_loaded():
+        (rc, out, err) = launchctl.unload()
+        if rc != 0:
+            launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_enabled(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if not launchctl.is_enabled():
+        (rc, out, err) = launchctl.enable()
+        if rc != 0:
+            launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_disabled(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if launchctl.is_enabled():
+        (rc, out, err) = launchctl.disable()
+        if rc != 0:
+            launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_started(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if not launchctl.is_loaded():
+        launchctl.module.fail_json(msg="attempted to start an unloaded job", label=launchctl.label)
+
+    if not launchctl.is_running():
+        (rc, out, err) = launchctl.start()
+        if rc != 0:
+            launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_stopped(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if not launchctl.is_loaded():
+        launchctl.module.fail_json(msg="attempted to stop an unloaded job", label=launchctl.label)
+        
+    if launchctl.is_running():
+        (rc, out, err) = launchctl.stop()
+        if rc != 0:
+            launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_restarted(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if not launchctl.is_loaded():
+        launchctl.module.fail_json(msg="attempted to restart an unloaded job", label=launchctl.label)
+        
+    (rc, out, err) = launchctl.restart()
+    if rc != 0:
+        launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)
+
+def handle_status(launchctl):
+    rc = None
+    out = ''
+    err = ''
+
+    if not launchctl.is_loaded():
+        launchctl.module.fail_json(msg="attempted to get the status of an unloaded job", label=launchctl.label)
+        
+    (rc, out, err) = launchctl.get_status()
+    if rc != 0:
+        launchctl.module.fail_json(msg=err, rc=rc, label=launchctl.label)
+
+    return (rc, out, err)    
+
+
+def main():
+    state_choices = ['loaded', 'unloaded', 'enabled', 'disabled', 'started', 'stopped', 'restarted', 'status']
+    module = AnsibleModule(
+        argument_spec = dict(
+            state=dict(default='loaded', choices=state_choices, type='str'),
+            domain=dict(required=True, type='str'),
+            label=dict(default=None, alias=['name'], type='str'),
+            path=dict(default=None, type='str'),
```

Comment:
make this a 'path' type and both ~ and $HOME as well as relative paths will be automatically resolved.